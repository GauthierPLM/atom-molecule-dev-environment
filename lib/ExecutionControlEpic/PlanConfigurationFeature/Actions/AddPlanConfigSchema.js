"use babel";
// @flow

import type {ConfigSchemaPart} from "../../../types";
import type {DevToolInfos} from "../Types/types.js.flow";
import type {AddPlanConfigSchemaAction} from "../Types/types";

export function addPlanConfigSchema(
  tool: DevToolInfos,
  schema: ConfigSchemaPart,
): AddPlanConfigSchemaAction {
  return {
    type: "ADD_PLAN_CONFIGURATION_SCHEMA",
    payload: {
      schema,
      tool,
    },
  };
}
