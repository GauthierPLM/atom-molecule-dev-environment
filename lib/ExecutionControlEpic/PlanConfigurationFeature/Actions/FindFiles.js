"use babel";
// @flow

import {
  refreshPackages,
} from "../../../ProjectSystemEpic/PackageFeature/Actions/RefreshPackages";
import type {
  PackageTester,
} from "../../../ProjectSystemEpic/PackageFeature/Types/types.js.flow";

export type FindFilesAction = (dispatch: (action: any) => void) => void;

export function findFiles(tester: PackageTester, id: string): FindFilesAction {
  return dispatch => {
    const rootPath = global.atom.project.getPaths()[0];
    let plugins = [
      {
        tool: { name: id, id: id, iconUri: "" },
        isPackage: tester,
      },
    ];
    dispatch(refreshPackages(rootPath, plugins));
  };
}
