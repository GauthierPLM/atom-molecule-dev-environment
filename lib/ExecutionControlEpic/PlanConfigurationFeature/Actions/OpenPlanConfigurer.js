"use babel";
// @flow

import type { DevToolInfos } from "../Types/types.js.flow";
import {
  renderToolConfigurer,
  renderToolPlans,
  toolPlansPanel,
} from "../AtomLinks/Panels";
import { sidePanel } from "../../../GlobalSystem/AtomLinks/Panels";
import {
  refreshPackages,
} from "../../../ProjectSystemEpic/PackageFeature/Actions/RefreshPackages";
import {
  DevToolsControllerInstance,
} from "../../../ExecutionControlEpic/DevtoolLoadingFeature/Model/DevToolsController";

export function openPlanConfigurer(tool: DevToolInfos) {
  return (dispatch: any) => {
    const plugins = DevToolsControllerInstance.getPackagesPlugins();
    renderToolConfigurer(sidePanel, tool);
    renderToolPlans(tool);
    toolPlansPanel.show();
    global.atom.project
      .getPaths()
      .map(projectPath => dispatch(refreshPackages(projectPath, plugins)));
  };
}
