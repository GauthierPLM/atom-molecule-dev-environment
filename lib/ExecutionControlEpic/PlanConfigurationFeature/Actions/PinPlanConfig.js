"use babel";
// @flow

import type {PlanConfig} from "../Types/types.js.flow";

export type PinPlanConfigAction = {
  type: "PIN_PLAN_CONFIGURATION",
  payload: {
    planConfig: PlanConfig,
  },
};

export function pinPlanConfig(planConfig: PlanConfig): PinPlanConfigAction {
  return {
    type: "PIN_PLAN_CONFIGURATION",
    payload: {
      planConfig,
    },
  };
}
