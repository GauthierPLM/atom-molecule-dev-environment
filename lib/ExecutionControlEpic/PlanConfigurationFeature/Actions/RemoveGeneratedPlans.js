"use babel";
// @flow

import type { DevToolInfo } from "../Types/types.js.flow";

export type RemoveGeneratedPlansAction = {
  type: "REMOVE_GENERATED_PLANS",
  payload: { tool: DevToolInfo, packagePath: string },
};

export function removeGeneratedPlans(
  tool: DevToolInfo,
  packagePath: string,
): RemoveGeneratedPlansAction {
  return {
    type: "REMOVE_GENERATED_PLANS",
    payload: { tool, packagePath },
  };
}
