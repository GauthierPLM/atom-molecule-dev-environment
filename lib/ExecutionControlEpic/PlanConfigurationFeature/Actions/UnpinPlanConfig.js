"use babel";
// @flow

import type { PlanConfig } from "../Types/types.js.flow";

export type UnpinPlanConfigAction = {
  type: "UNPIN_TARGET_CONFIGURATION",
  payload: {
    planConfig: mixed
  }
};

export function unpinPlanConfig(planConfig: PlanConfig): UnpinPlanConfigAction {
  return {
    type: "UNPIN_TARGET_CONFIGURATION",
    payload: {
      planConfig
    }
  };
}
