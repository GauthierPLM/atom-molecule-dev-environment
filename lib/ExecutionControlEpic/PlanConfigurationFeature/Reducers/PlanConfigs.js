"use babel";
// @flow

import type {PlanConfig} from "../Types/types.js.flow";
import {fromJS, is} from "immutable";

export default function(
  state: PlanConfigsReducer = [],
  action: any,
): PlanConfigsReducer {
  switch (action.type) {
    case "ADD_PLAN_CONFIGURATION":
      return fromJS(state)
        .push({
          name: action.payload.name,
          tool: action.payload.tool,
          config: action.payload.config,
          pinned: false,
          stager: action.payload.stager,
          packageInfos: action.payload.packageInfos,
        })
        .toJS();
    case "PIN_PLAN_CONFIGURATION":
      return fromJS(state)
        .map(
          plan =>
            is(plan, fromJS(action.payload.planConfig).remove("state"))
              ? plan.set("pinned", true)
              : plan,
        )
        .toJS();
    case "UNPIN_PLAN_CONFIGURATION":
      return fromJS(state)
        .map(
          plan =>
            is(plan, fromJS(action.payload.planConfig).remove("state"))
              ? plan.set("pinned", false)
              : plan,
        )
        .toJS();
    case "REMOVE_PLAN_CONFIGURATION":
      return fromJS(state)
        .filter(
          plan =>
            is(plan, fromJS(action.payload.planConfig).remove("state"))
              ? false
              : plan,
        )
        .toJS();
    default:
      return state;
  }
}

export type PlanConfigsReducer = Array<PlanConfig>;
