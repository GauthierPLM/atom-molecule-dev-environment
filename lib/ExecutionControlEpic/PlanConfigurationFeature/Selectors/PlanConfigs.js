'use babel'
// @flow

import type {PlanConfig} from "../Types/types.js.flow";
import type {PlanConfigsReducer} from "../Reducers/PlanConfigs";

export function selectPlansOfTool(state: PlanConfigsReducer, toolId: string): Array<PlanConfig> {
  return state.filter(plan => plan.tool.id == toolId);
};

export function selectPinnedPlans(state: PlanConfigsReducer): Array<PlanConfig> {
  return state.filter(plan => plan.pinned);
};
