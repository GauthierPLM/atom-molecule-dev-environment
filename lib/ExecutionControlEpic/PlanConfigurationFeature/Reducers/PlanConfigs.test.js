"use babel";
// @flow

import PlanConfigs from "./PlanConfigs";
import type { AddPlanConfigAction } from "../Actions/AddPlanConfig";
import { addPlanConfig } from "../Actions/AddPlanConfig";
import type { PinPlanConfigAction } from "../Actions/PinPlanConfig";
import { pinPlanConfig } from "../Actions/PinPlanConfig";
import type { UnpinPlanConfigAction } from "../Actions/UnpinPlanConfig";
import { unpinPlanConfig } from "../Actions/UnpinPlanConfig";
import type { RemovePlanConfigAction } from "../Actions/RemovePlanConfig";
import { removePlanConfig } from "../Actions/RemovePlanConfig";
import type { PlanConfig } from "../Types/types.js.flow";

describe("PlanConfigs", () => {
  describe("ADD_PLAN_CONFIGURATION", () => {
    it("should add a plan configuration", () => {
      let state = [];
      let action: AddPlanConfigAction = addPlanConfig(
        "ls",
        {
          id: "toolid",
          name: "shell",
          iconUri: "atom://myplugin/icon.png",
        },
        "ls -l",
        { type: "integrated" },
        {
          name: "",
          path: "/",
          type: "directory",
        },
      );
      let subject = PlanConfigs(state, action);

      expect(subject).toMatchSnapshot();
    });
  });

  describe("PIN_PLAN_CONFIGURATION", () => {
    it("should pin plan config", () => {
      let planConfig: PlanConfig = {
        name: "ls",
        tool: {
          id: "toolid",
          name: "shell",
          iconUri: "atom://myplugin/icon.png",
        },
        config: "ls -l",
        stager: { type: "integrated" },
        packageInfos: {
          name: "",
          path: "/",
          type: "directory",
        },
        pinned: false,
      };
      let planConfig2: PlanConfig = {
        name: "mimikatz",
        tool: {
          id: "toolid",
          name: "powershell",
          iconUri: "atom://myplugin/icon.png",
        },
        config: "Invoke-Mimikatz",
        stager: { type: "integrated" },
        packageInfos: {
          name: "",
          path: "/",
          type: "directory",
        },
        pinned: false,
      };
      let state = [planConfig, planConfig2];
      let action: PinPlanConfigAction = pinPlanConfig(
        Object.assign({}, planConfig, { state: null }),
      );
      let subject = PlanConfigs(state, action);

      expect(subject).toMatchSnapshot();
    });
  });

  describe("UNPIN_PLAN_CONFIGURATION", () => {
    it("should unpin plan config", () => {
      let planConfig: PlanConfig = {
        name: "ls",
        tool: {
          id: "toolid",
          name: "shell",
          iconUri: "atom://myplugin/icon.png",
        },
        config: "ls -l",
        stager: { type: "integrated" },
        packageInfos: {
          name: "",
          path: "/",
          type: "directory",
        },
        pinned: true,
      };
      let planConfig2: PlanConfig = {
        name: "mimikatz",
        tool: {
          id: "toolid",
          name: "powershell",
          iconUri: "atom://myplugin/icon.png",
        },
        config: "Invoke-Mimikatz",
        stager: { type: "integrated" },
        packageInfos: {
          name: "",
          path: "/",
          type: "directory",
        },
        pinned: false,
      };
      let state = [planConfig, planConfig2];
      let action: UnpinPlanConfigAction = unpinPlanConfig(
        Object.assign({}, planConfig, { state: null }),
      );
      let subject = PlanConfigs(state, action);

      expect(subject).toMatchSnapshot();
    });
  });

  describe("REMOVE_PLAN_CONFIGURATION", () => {
    it("should remove plan config", () => {
      let planConfig: PlanConfig = {
        name: "ls",
        tool: {
          id: "toolid",
          name: "shell",
          iconUri: "atom://myplugin/icon.png",
        },
        config: "ls -l",
        stager: { type: "integrated" },
        packageInfos: {
          name: "",
          path: "/",
          type: "directory",
        },
        pinned: false,
      };
      let planConfig2: PlanConfig = {
        name: "mimikatz",
        tool: {
          id: "toolid",
          name: "powershell",
          iconUri: "atom://myplugin/icon.png",
        },
        config: "Invoke-Mimikatz",
        stager: { type: "integrated" },
        packageInfos: {
          name: "",
          path: "/",
          type: "directory",
        },
        pinned: false,
      };
      let state = [planConfig, planConfig2];
      let action: RemovePlanConfigAction = removePlanConfig(
        Object.assign({}, planConfig, { state: null }),
      );
      let subject = PlanConfigs(state, action);

      expect(subject).toMatchSnapshot();
    });
  });
});
