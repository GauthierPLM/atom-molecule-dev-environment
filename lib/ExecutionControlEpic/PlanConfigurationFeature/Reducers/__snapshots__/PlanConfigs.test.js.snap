// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanConfigs ADD_PLAN_CONFIGURATION should add a plan configuration 1`] = `
Array [
  Object {
    "config": "ls -l",
    "name": "ls",
    "packageInfos": undefined,
    "pinned": false,
    "stager": Object {
      "type": "integrated",
    },
    "tool": Object {
      "iconUri": "atom://myplugin/icon.png",
      "id": "toolid",
      "name": "shell",
    },
  },
]
`;

exports[`PlanConfigs PIN_PLAN_CONFIGURATION should pin plan config 1`] = `
Array [
  Object {
    "config": "ls -l",
    "name": "ls",
    "packageInfos": Object {
      "name": "",
      "path": "/",
      "type": "directory",
    },
    "pinned": true,
    "stager": Object {
      "type": "integrated",
    },
    "tool": Object {
      "iconUri": "atom://myplugin/icon.png",
      "id": "toolid",
      "name": "shell",
    },
  },
  Object {
    "config": "Invoke-Mimikatz",
    "name": "mimikatz",
    "packageInfos": Object {
      "name": "",
      "path": "/",
      "type": "directory",
    },
    "pinned": false,
    "stager": Object {
      "type": "integrated",
    },
    "tool": Object {
      "iconUri": "atom://myplugin/icon.png",
      "id": "toolid",
      "name": "powershell",
    },
  },
]
`;

exports[`PlanConfigs REMOVE_PLAN_CONFIGURATION should remove plan config 1`] = `
Array [
  Object {
    "config": "Invoke-Mimikatz",
    "name": "mimikatz",
    "packageInfos": Object {
      "name": "",
      "path": "/",
      "type": "directory",
    },
    "pinned": false,
    "stager": Object {
      "type": "integrated",
    },
    "tool": Object {
      "iconUri": "atom://myplugin/icon.png",
      "id": "toolid",
      "name": "powershell",
    },
  },
]
`;

exports[`PlanConfigs UNPIN_PLAN_CONFIGURATION should unpin plan config 1`] = `
Array [
  Object {
    "config": "ls -l",
    "name": "ls",
    "packageInfos": Object {
      "name": "",
      "path": "/",
      "type": "directory",
    },
    "pinned": false,
    "stager": Object {
      "type": "integrated",
    },
    "tool": Object {
      "iconUri": "atom://myplugin/icon.png",
      "id": "toolid",
      "name": "shell",
    },
  },
  Object {
    "config": "Invoke-Mimikatz",
    "name": "mimikatz",
    "packageInfos": Object {
      "name": "",
      "path": "/",
      "type": "directory",
    },
    "pinned": false,
    "stager": Object {
      "type": "integrated",
    },
    "tool": Object {
      "iconUri": "atom://myplugin/icon.png",
      "id": "toolid",
      "name": "powershell",
    },
  },
]
`;
